packages/reactive-element/src/decorators/query-assigned-elements.ts

```diff
@@ -93,5 +97,13 @@ export function queryAssignedElements(
         ) as V;
       },
     };
+    if (typeof name !== 'object' && extendedReflect.decorate) {
```

Comment:
I think you could even wrap this into a function exported from base, like:

base.ts:
```ts
/**
  * Ensures that the property is defined even if we're called as a legacy decorator, and
  *  Reflect.decorate is present. In that case we have no guarantees that a descriptor 
  * returned from a decorator will be
  * defined on the class, so we must apply it directly ourselves.
  * Also applies defaults...
  */
export const ensureProperty = (obj, nameOrContext, descriptor) => {
  descriptor = {
      enumerable: true,
      configurable: true,
      ...descriptor,
  };
  if (typeof name !== 'object' && extendedReflect.decorate) {
      Object.defineProperty(obj, name, descriptor);
      return;
  }
  return descriptor;
}
```

Here:
```ts
return ensureProperty(obj, name, descriptor);
```

Reply:
Done. Gave it a short name for size and because it has no easily explicable name, and just mutated the descriptor rather than allocating a new one, since we know all the cases where this is called and they're all ok with mutation